pkg/ebpf/tracee.go

```diff
@@ -1124,39 +1119,38 @@ var netSeqOps = [6]string{
 	"raw6_seq_ops",
 }
 
-func (t *Tracee) invokeIoctlTriggeredEvents(cmds int32) error {
-	// invoke HookedSyscallsEvent
-	if cmds&IoctlFetchSyscalls == IoctlFetchSyscalls {
-		_, ok := t.events[events.HookedSyscalls]
-		if ok {
-			ptmx, err := os.OpenFile(t.config.Capture.OutputPath, os.O_RDONLY, 0444)
-			if err != nil {
-				return err
-			}
-			syscall.Syscall(syscall.SYS_IOCTL, ptmx.Fd(), uintptr(IoctlFetchSyscalls), 0)
-			ptmx.Close()
-		}
+func (t *Tracee) triggerSyscallsIntegrityCheck() {
```

Comment:
Alright, let's be honest here.. the previous function `invokeIoctlTriggeredEvents()` already existed as a function for `*Tracee` so I'm willing to accept/agree having these functions here for now, but this should be **temporary** as the functions are unrelated to `*Tracee` and related to specific events.

If we compare all other `(*Tracee).Functions` we have "initBPF", "getOptionsConfig", "generateInitValues", ... and they're all generic to functioning of Tracee (not to specific events), or they should be.

We should probably create an interface for anything relying in similar approach(es) (anything that needs a trigger + derived events + re-trigger, based in uprobes and things like that). @NDStrahilevitz for the record here (as he is our refactoring guy =D).

I'm willing to move forward as we're basically replacing the ioctl() mechanism for uprobe, but I think it needs refactor in the next version (and an issue opened).

Reply:
Yep, please add a TODO to move away from this.

Reply:
Sure I agree, I'll add a TODO and add that to #2055